src/riakc_ts.erl

```diff
@@ -37,91 +37,75 @@
 
--spec query(Pid::pid(), Query::string(), Interpolations::[{binary(), binary()}], Cover::term()) ->
+-spec query(pid(), Query::string(), Interpolations::[{binary(), binary()}], Cover::term()) ->
                    {ColumnNames::[binary()], Rows::[tuple()]} | {error, term()}.
-%% @doc Execute a "SELECT ..." Query with client.  The result returned
+%% @doc Execute a Query with client.  The result returned
 %%      is a tuple containing a list of columns as binaries in the
 %%      first element, and a list of records, each represented as a
 %%      list of values, in the second element, or an @{error, Reason@}
 %%      tuple.
-
-query(Pid, QueryText, Interpolations, Cover) ->
-    Message = riakc_ts_query_operator:serialize(QueryText, Interpolations),
-    Response = server_call(Pid, Message#tsqueryreq{cover_context=Cover}),
+query(Pid, Query, Interpolations, Cover)
+  when is_pid(Pid), is_list(Query) ->
+    Message = riakc_ts_query_operator:serialize(
+                iolist_to_binary(Query), Interpolations),
+    Response = server_call(Pid, Message#tsqueryreq{cover_context = Cover}),
     riakc_ts_query_operator:deserialize(Response).
 
+
+-spec get_coverage(pid(), table_name(), Query::string()) ->
+                          {ok, Entries::[term()]} | {error, term()}.
 %% @doc Generate a parallel coverage plan for the specified query
-get_coverage(Pid, Table, QueryText) ->
+get_coverage(Pid, Table, Query)
+  when is_pid(Pid), (is_binary(Table) orelse is_list(Table)), is_list(Query) ->
     server_call(Pid,
-                #tscoveragereq{query = #tsinterpolation{base=QueryText},
-                               replace_cover=undefined,
-                               table = Table}).
+                #tscoveragereq{query = #tsinterpolation{base = iolist_to_binary(Query)},
+                               replace_cover = undefined,
+                               table = iolist_to_binary(Table)}).
 
--spec put(Pid::pid(), Table::table_name(), [[ts_value()]]) ->
-                 ok | {error, Reason::term()}.
-%% @doc Make data records from Data and insert them, individually,
-%%      into a time-series Table, using client Pid. Each record is a
-%%      list of values of appropriate types for the complete set of
-%%      table column names, in the order in which they appear in table's
-%%      DDL.  On success, 'ok' is returned, else an @{error, Reason@}
-%%      tuple.  Also @see put/3.
-%%
-%%      As of 2015-11-05, ColumnNames parameter is ignored, the function
-%%      expects the full set of fields in each element of Data.
-%%
 
-put(Pid, TableName, Measurements) ->
-    put(Pid, TableName, [], Measurements).
+-spec put(pid(), table_name(), [[ts_value()]]) ->
+                 ok | {error, Reason::term()}.
+%% @equiv put/4.
+put(Pid, Table, Measurements) ->
+    put(Pid, Table, [], Measurements).
 
--spec put(Pid::pid(), Table::table_name(), ColumnNames::[ts_columnname()], Data::[[ts_value()]]) ->
```

Comment:
Why remove the column names? Seems like we should be consistent across the call: either all names or just types.